apiserver/storage/storage.go

```diff
@@ -200,3 +201,69 @@ func (a *API) checkProviderTypeSupported(providerType storage.ProviderType) erro
 	}
 	return nil
 }
+
+var getVolumeManager = func(vs volume.VolumeState) volume.VolumeManager {
+	return volume.NewVolumeManager(vs)
+}
+
+func (a *API) ListVolumes(filter params.StorageVolumeFilter) (params.StorageVolumesResult, error) {
+	volumeManager := getVolumeManager(a.storage)
+
+	all, err := volumeManager.List()
+	if err != nil {
+		return params.StorageVolumesResult{}, err
+	}
+
+	if len(all) < 1 {
```

Comment:
you can't overflow a container size, so len will always be >= 0. it's signed so it can be used in for loops and have less surprising behaviour when you do overflow the index